lib/install/deps.js

```diff
@@ -600,6 +600,7 @@ var earliestInstallable = exports.earliestInstallable = function (requiredBy, tr
   var binaryMatches = typeof pkg.bin === 'object' && tree.children.some(function (child) {
     if (child.removed) return false
     if (typeof child.package.bin !== 'object') return false
+    if (!pkg.bin || !child.package.bin) return false
```

Comment:
The only values that are supposed to come out of `read-package-json`/`normalize-package-json` are an array or undefined/null. (If you had a string in your `package.json` that gets converted to an object before this code can see it.)

Truthyness is sufficient for `pkg.bin` but `child.package.bin` needs truthyness (or not null) checking AND object-ness-checking as it's gonna call `Object.keys`.

My inclination is to have the guard be the same for both fields.